src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/PerpetualPositionsProcessor.kt

```diff
@@ -1,13 +1,71 @@
 package exchange.dydx.abacus.processor.wallet.account
 
 import exchange.dydx.abacus.processor.base.BaseProcessor
+import exchange.dydx.abacus.protocols.LocalizerProtocol
 import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.internalstate.InternalPerpetualPosition
 import exchange.dydx.abacus.utils.modify
 import exchange.dydx.abacus.utils.mutable
 import exchange.dydx.abacus.utils.safeSet
+import indexer.codegen.IndexerPerpetualPositionResponseObject
 
-internal class PerpetualPositionsProcessor(parser: ParserProtocol) : BaseProcessor(parser) {
-    private val itemProcessor = PerpetualPositionProcessor(parser = parser)
+internal class PerpetualPositionsProcessor(
+    parser: ParserProtocol,
+    localizer: LocalizerProtocol?,
+    private val itemProcessor: PerpetualPositionProcessorProtocol =
+        PerpetualPositionProcessor(parser = parser, localizer = localizer),
+) : BaseProcessor(parser) {
+
+    fun process(
+        existing: Map<String, InternalPerpetualPosition>? = null,
+        payload: Map<String, IndexerPerpetualPositionResponseObject>?,
+    ): Map<String, InternalPerpetualPosition>? {
+        return if (payload != null) {
+            val result = mutableMapOf<String, InternalPerpetualPosition>()
+            for ((key, value) in payload.entries) {
+                val existingPosition = existing?.get(key)
+                val newPosition = itemProcessor.process(existingPosition, value)
+                if (newPosition != null) {
+                    result[key] = newPosition
+                } else {
+                    result.remove(key)
+                }
+            }
+            return if (result != existing) {
+                result
+            } else {
+                existing
+            }
+        } else {
+            existing
+        }
+    }
+
+    fun processChanges(
+        existing: Map<String, InternalPerpetualPosition>?,
+        payload: List<IndexerPerpetualPositionResponseObject>?,
+    ): Map<String, InternalPerpetualPosition>? {
+        return if (payload != null) {
```

Comment:
[non-blocking] easier for me to read if we use a guard clause 🙏 . also, wouldn't 
```
            return if (result != existing) {
                result
            } else {
                existing
            }
```
be the same as
```
return result
```
since if result is == to existing, it's the same thing as returning existing? or is there a referential equality check somewhere that we need to fulfill?

```suggestion
        if (payload == null) return existing
        val result = existing?.toMutableMap() ?: mutableMapOf()
        for (item in payload) {
                if (item.market != null) {
                    val newPosition = itemProcessor.processChanges(result[item.market], item)
                    if (newPosition != null) {
                        result[item.market] = newPosition
                    } else {
                        result.remove(item.market)
                    }
                }
            }
            return result
```

Reply:
It's copied from the old code, but I don't think it's really needed at the processor level.  I will update them in follow-up PRs.